src/lib.rs

```diff
@@ -61,9 +65,116 @@ impl State {
     }
 }
 
+#[cfg(feature="std")]
+mod oscbuf {
+    use super::MAX_OSC_RAW;
+    pub const DEFAULT_SIZE_LIMIT: usize = u32::max_value() as usize;
+
+    /// When std is available, we can simply use a Vec for the
+    /// OSC storage.  We'll pre-size it to something reasonable.
+    pub struct OscBuffer {
+        buf: Vec<u8>,
+        size_limit: usize,
+    }
+
+    impl std::ops::Deref for OscBuffer {
+        type Target = Vec<u8>;
+        fn deref(&self) -> &Vec<u8> {
+            &self.buf
+        }
+    }
+
+    impl std::ops::DerefMut for OscBuffer {
+        fn deref_mut(&mut self) -> &mut Vec<u8> {
+            &mut self.buf
+        }
+    }
+
+    impl OscBuffer {
+        pub fn new(size_limit: usize) -> Self {
+            Self {
+                buf: Vec::with_capacity(MAX_OSC_RAW.min(size_limit)),
+                size_limit,
+            }
+        }
+
+        pub fn clear(&mut self) {
+            // If we're a bit heavy, shrink our usage.
+            // Ideally we'd call Vec::shrink_to() but that is
+            // nightly only, so we fake it by resizing and then
+            // asking for a shrink to fit the desired capacity.
+            if self.buf.capacity() > 2 * MAX_OSC_RAW {
+                self.buf.resize(MAX_OSC_RAW, 0u8);
+                self.buf.shrink_to_fit();
+            }
+            self.buf.clear();
+        }
+
+        /// Returns true if we are able to store an additional
+        /// byte of data.  Since we have an allocator available,
+        /// we always return true.
+        pub fn have_capacity(&self) -> bool {
+            self.buf.len() < self.size_limit
+        }
+    }
+}
+
+#[cfg(not(feature="std"))]
+mod oscbuf {
+    use super::MAX_OSC_RAW;
+    pub const DEFAULT_SIZE_LIMIT: usize = MAX_OSC_RAW;
+    /// The no_std version of OscBuffer uses a statically allocated
+    /// array and thus has a limit on the size of the data that it
+    /// can hold.  If an OSC exceeds this length, it will be silently
+    /// truncated.
+    /// OscBuffer presents a subset of the Vec API.
```

Comment:
Almost seems like [`FixedVec`](https://docs.rs/fixedvec/0.2.3/fixedvec) could be nice, but it looks like it might not be better, or at least not yet.